daemon/image_delete.go

```diff
@@ -93,27 +96,38 @@ func (daemon *Daemon) imgDeleteHelper(name string, list *[]types.ImageDelete, fi
 		return nil
 	}
 
-	if len(repos) <= 1 || (len(repoAndTags) <= 1 && deleteByID) {
-		if err := daemon.canDeleteImage(img.ID, force); err != nil {
-			return err
+	// Untag the current image
+	untag := func(repoAndTags map[string][]string) error {
+		for repoName, tags := range repoAndTags {
+			for _, tag := range tags {
+				tagDeleted, err := daemon.Repositories().Delete(repoName, tag)
+				if err != nil {
+					return err
+				}
+				if tagDeleted {
+					*list = append(*list, types.ImageDelete{
+						Untagged: utils.ImageReference(repoName, tag),
+					})
+					daemon.EventsService.Log("untag", img.ID, "")
+				}
+			}
 		}
+		return nil
 	}
 
-	// Untag the current image
-	for repoName, tags := range repoAndTags {
-		for _, tag := range tags {
-			tagDeleted, err := daemon.Repositories().Delete(repoName, tag)
-			if err != nil {
+	if len(repos) <= 1 || (len(repoAndTags) <= 1 && deleteByID) {
```

Comment:
This was not added by me, but I think adding a comment is really necessary.
When I was trying to find out why this line is here, I seemed to find another potential unexpected behavior. see https://github.com/docker/docker/issues/14116 and https://github.com/mountkin/docker/commit/debbb1f968f770715a44806991b5efaace132000